cloudbuild.yaml

```
@@ -1,10 +1,25 @@
 steps:
-  - name: "gcr.io/cloud-builders/docker"
-    entrypoint: "/bin/bash"
-    args:
-      - "-c"
-      - |
-        docker pull gcr.io/${PROJECT_ID}/linear-airbyte-source:latest
-        docker build -t gcr.io/${PROJECT_ID}/linear-airbyte-source:$BUILD_ID -t gcr.io/${PROJECT_ID}/linear-airbyte-source:latest --cache-from gcr.io/${PROJECT_ID}/linear-airbyte-source:latest --build-arg BUILDKIT_INLINE_CACHE=1 -f Dockerfile .
-        [[ "$BRANCH_NAME" == "main" ]] && docker push gcr.io/${PROJECT_ID}/linear-airbyte-source:$BUILD_ID || true
-        [[ "$BRANCH_NAME" == "main" ]] && docker push gcr.io/${PROJECT_ID}/linear-airbyte-source:latest || true
+- name: 'gcr.io/cloud-builders/docker:20.10.3'
+  args: ['run', '--privileged', 'linuxkit/binfmt:v0.7']
+  id: 'initialize-qemu'
+- name: 'gcr.io/cloud-builders/docker:20.10.3'
+  args: ['buildx', 'create', '--name', 'mybuilder']
+  id: 'create-builder'
+- name: 'gcr.io/cloud-builders/docker:20.10.3'
+  args: ['buildx', 'use', 'mybuilder']
+  id: 'select-builder'
+- name: 'gcr.io/cloud-builders/docker:20.10.3'
+  args: ['buildx', 'inspect', '--bootstrap']
+  id: 'show-target-build-platforms'
+- name: 'gcr.io/cloud-builders/docker:20.10.3'
+  entrypoint: "/bin/bash"
+  args:
+    - "-c"
+    - |
+      [[ "$BRANCH_NAME" == "main" ]] && docker buildx build --platform=$_DOCKER_BUILDX_PLATFORMS -t gcr.io/${PROJECT_ID}/linear-airbyte-source:$BUILD_ID -t gcr.io/${PROJECT_ID}/linear-airbyte-source:latest --push -f Dockerfile . || docker buildx build --platform=$_DOCKER_BUILDX_PLATFORMS -t gcr.io/${PROJECT_ID}/linear-airbyte-source:$BUILD_ID -t gcr.io/${PROJECT_ID}/linear-airbyte-source:latest -f Dockerfile .
```

Just split this up into two cloudbuild tasks. one that builds. and one that pushes conditionally.  This should work, no?
```suggestion
      docker buildx build --platform=$_DOCKER_BUILDX_PLATFORMS -t gcr.io/${PROJECT_ID}/linear-airbyte-source:$BUILD_ID -t gcr.io/${PROJECT_ID}/linear-airbyte-source:latest -f Dockerfile .
      [[ "$BRANCH_NAME" == "main" ]] && docker push gcr.io/${PROJECT_ID}/linear-airbyte-source:latest || echo "Only built."
```

---

no this does not work with multi platform builds. when using `buildx` you can either specify `--push` which will push built images and the manifest or `--load`, which will load built images into docker namespace in order to push them. however there is a bug with `--load` that it does not work when multiple platforms are used 🤷

---

Ah, keep it as is then...